supportedBackends/htcondor/src/main/scala/cromwell/backend/impl/htcondor/HtCondorJobExecutionActor.scala

```diff
@@ -68,6 +71,30 @@ class HtCondorJobExecutionActor(override val jobDescriptor: BackendJobDescriptor
     HtCondorRuntimeAttributes(runtimeMap, jobDescriptor.descriptor.workflowOptions)
   }
 
+  private val cacheActor = context.actorOf(cacheActorProps, s"CacheActor-${jobDescriptor.call.fullyQualifiedName}")
+
+  log.debug("{} Calculating hash for current job.", tag)
+  private val jobHash = calculateHash
+
+  private val executionResponse = Promise[BackendJobExecutionResponse]()
+
+  override def receive = super.receive orElse {
+    case ExecutionResultFound(succeededResponse) => executionResponse success succeededResponse.copy(jobKey = jobDescriptor.key)
+
+    case ExecutionResultNotFound =>
+      createExecutionFolderAndScript
+      executionResponse success executeTask()
+
+    case ExecutionResultStored(hash) => log.debug("{} Cache entry was stored for Job with hash {}.", tag, hash)
+
+    case ExecutionResultAlreadyExist => log.warning("{} Cache entry for hash {} already exist.", tag, jobHash)
+
+    case message: Any =>
+      val msg = s"Unhandled message received. Message: $message."
+      log.error(msg)
+      FailedNonRetryableResponse(jobDescriptor.key, new Throwable(msg), None)
```

Comment:
How do we distinguish this response with Retryable response ? I mean is there a separate mechanism for retry and sending corresponding failure response to it?